src/sqlfluff/dialects/dialect_clickhouse.py

```diff
@@ -1421,6 +1421,38 @@ class DropUserStatementSegment(ansi.DropUserStatementSegment):
     )
 
 
+class CreateUserStatementSegment(BaseSegment):
+    """A `CREATE USER` statement.
+
+    As specified in
+    https://clickhouse.com/docs/en/sql-reference/statements/create/user/
+    """
+
+    type = "create_user_statement"
+
+    match_grammar = Sequence(
+        "CREATE",
+        "USER",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("SingleIdentifierGrammar"),
+        Ref("OnClusterClauseSegment", optional=True),
+        # IDENTIFIED BY 'password' or IDENTIFIED WITH ... BY ...
+        Sequence(
+            "IDENTIFIED",
+            OneOf(
+                Sequence(
+                    "WITH",
+                    Ref("SingleIdentifierGrammar"),
+                    "BY",
+                    Ref("QuotedLiteralSegment"),
+                ),
+                Sequence("BY", Ref("QuotedLiteralSegment")),
+            ),
+            optional=True,
```

Comment:
This can be simplified here.
```suggestion
            "IDENTIFIED",
            Sequence("WITH", Ref("SingleIdentifierGrammar"), optional=True),
            Sequence("BY", Ref("QuotedLiteralSegment")),
            optional=True,
```